client/my-sites/checkout/checkout-thank-you/index.jsx

```diff
@@ -403,6 +404,11 @@ export class CheckoutThankYou extends Component {
 			);
 			wasDIFMProduct = purchases.some( isDIFMProduct );
 			wasTitanEmailOnlyProduct = purchases.length === 1 && purchases.some( isTitanMail );
+			wasDomainOnly =
+				selectedSite?.options?.is_domain_only &&
+				purchases.every(
+					( purchase ) => isDomainMapping( purchase ) || isDomainRegistration( purchase )
```

Comment:
In my testing, `purchases` contained one domain mapping product and one domain registration product after finishing a domain only purchase.

Reply:
I realized that this doesn't impact the intended behaviour so I left it out.